dask/array/linalg.py

```diff
@@ -167,6 +166,79 @@ def tsqr(data, name=None, compute_svd=False):
         return u, s, v
 
 
+def compression_level(n, q):
+    return min(max(20, q + 10), n)
+
+
+def compression_matrix(data, q, n_power_iter=0):
+    """ Uses random sampling to identify a subspace that captures most
+    of the action of a matrix.
+
+    As presented in:
+
+        A. Benson, D. Gleich, and J. Demmel.
+        Finding structure with randomness: Probabilistic algorithms for
+        constructing approximate matrix decompositions.
+        SIAM Rev., Survey and Review section, Vol. 53, num. 2,
+        pp. 217-288, June 2011
+        http://arxiv.org/abs/0909.4061
+
+    This compression matrix returned by this algorithm can be used to
+    compute both the QR decomposition and the Singular Value
+    Decomposition.
+
+    Parameters
+    ----------
+
+    data: Array
+    q: Size of the desired subspace (the actual size will be bigger,
+    because of oversampling)
+    n_power_iter: number of power iterations, useful when the singular
+    values of the input matrix decay very slowly.
+    """
+    n = data.shape[1]
+    comp_level = compression_level(n, q)
+    omega = standard_normal(size=(n, comp_level), chunks=(data.chunks[1],
+                                                          (comp_level,)))
+    mat_h = data.dot(omega)
+    for j in range(n_power_iter):
+        mat_h = data.dot(data.T.dot(mat_h))
```

Comment:
Often one evolves from a power method to something that periodically rescales a bit in order to avoid losing the less powerful singular vectors.  Should we be concerned about this here?